sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClientBuilder.java

```diff
@@ -294,18 +325,22 @@ public HttpClient build() {
      * If the timeout is {@code null} the default timeout will be used. If the timeout is less than or equal to zero
      * no timeout will be used. If the timeout is less than one millisecond a timeout of one millisecond will be used.
      */
-    static long getTimeoutMillis(Duration configuredTimeout, long defaultTimeout) {
+    static Duration getTimeout(Duration configuredTimeout, Duration defaultTimeout) {
         // Timeout is null, use the default timeout.
         if (configuredTimeout == null) {
             return defaultTimeout;
         }
 
         // Timeout is less than or equal to zero, return no timeout.
         if (configuredTimeout.isZero() || configuredTimeout.isNegative()) {
-            return 0;
+            return Duration.ZERO;
         }
 
         // Return the maximum of the timeout period and the minimum allowed timeout period.
-        return Math.max(configuredTimeout.toMillis(), MINIMUM_TIMEOUT);
+        if (configuredTimeout.compareTo(MINIMUM_TIMEOUT) < 0) {
+            return MINIMUM_TIMEOUT;
```

Comment:
Do we need to log this ?

Reply:
I don't think so, it has not been in previous version of this code. This behavior is also documented on OkHttpAsyncHttpClientBuilder methods.